packages/wrangler/src/deploy/deploy.ts

```diff
@@ -1068,25 +1053,23 @@ export async function updateQueueConsumers(
 				);
 				continue;
 			}
-
-			const body: PostTypedConsumerBody = {
-				type: consumer.type,
-				dead_letter_queue: consumer.dead_letter_queue,
-				settings: {
-					batch_size: consumer.max_batch_size,
-					max_retries: consumer.max_retries,
-					visibility_timeout_ms: consumer.visibility_timeout_ms,
-					retry_delay: consumer.retry_delay,
-				},
-			};
 			updateConsumers.push(
-				postTypedConsumer(config, consumer.queue, body).then(() => [
+				postConsumer(config, consumer.queue, body).then(() => [
 					`Consumer for ${consumer.queue}`,
 				])
 			);
 		} else {
-			const body: PutConsumerBody = {
+			if (config.name === undefined) {
+				// TODO: how can we reliably get the current script name?
+				throw new UserError(
+					"Script name is required to update queue consumers"
+				);
+			}
+
+			const body: PostTypedConsumerBody = {
+				type: "worker",
```

Comment:
Q: should we still map this to `consumer.type`, and set `type=worker` as the default in the toml parser?